### DISCUSSION TITLE
Invalid or unexpected token for images inside node-modules/

### DISCUSSION THREAD

**Discussion by moaaz-bhnas** [vercel/next.js#12410] [Help]
I use an npm module ([`react-images-upload`](https://www.npmjs.com/package/react-images-upload)) that uses an SVG icon internally. The package works fine with the svg icon, unless I reload the specific page where that pachage is used, or write its url directly.   

The error:
`Unexpected token '<'`  

I thought it might be a problem with the svg file itself, so I tried to replace the `.svg` file in the package with a `.png` file, but still gives me an error when I reload the page.

## To Reproduce

Repo: https://github.com/moaaz-bhnas/hdomi

1. Clone the repo: `git clone https://github.com/moaaz-bhnas/hdomi.git`
2. Install required packages: `npm install`
3. run: `npm run dev`
4. Write the url direclty for the page that uses the package: [http://localhost:3000/seller/add-product](http://localhost:3000/seller/add-product)

## Expected behavior

The package displays with the icon when I reload the page or enter its url directly in search tab.

## Screenshots

![Screenshot](https://user-images.githubusercontent.com/34780298/80865772-bf892380-8c8b-11ea-8533-bb4cf2e30a07.png)

## System information

- OS: Windows
- Browser: Chrome
- Version of Next.js: latest

**Comment by lboecker** [comment]
You have to tell webpack how to load SVG files. See:

https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config
https://webpack.js.org/loaders/svg-inline-loader/

**Comment by moaaz-bhnas** [comment]
@lboecker   
I have this as my webpack config:  
```
module.exports = {
  webpack(config) {
    // url loader
    config.module.rules.push({
      test: /\.(png|jpg|gif|svg)$/,
      use: {
        loader: 'url-loader'
      }
    });

    return config;
  },
}

```

**Comment by lboecker** [comment]
Try to explicitly include the directory like so:
```js
const path = require('path');

module.exports = {
  webpack(config) {
    // url loader
    config.module.rules.push({
      test: /\.(png|jpg|gif|svg)$/,
      include: [path.resolve(__dirname, 'node_modules/react-images-upload/src/component')]
      use: {
        loader: 'url-loader'
      }
    });

    return config;
  },
}
```

**Comment by moaaz-bhnas** [comment]
@lboecker  
I tried this path as the svg icon lives directly inside the package (`node_modules/react-images-upload/icon.svg`):  

```
const path = require('path');

module.exports = {
  webpack(config) {
    // url loader
    config.module.rules.push({
      test: /\.(png|jpg|gif|svg)$/,
      include: [path.resolve(__dirname, 'node_modules/react-images-upload')],
      use: {
        loader: 'url-loader'
      }
    });

    return config;
  },
}
```

But somehow the loader doesn't work at all, even for images in my `src` folder. I'm unfortunately not good with webpack and configs ..

**Comment by lboecker** [comment]
Sorry, I got the path wrong. Since your Next.js project lives in a subdirectory, the path should look like this: `path.resolve(__dirname, '../../node_modules/react-images-upload')`.

> But somehow the loader doesn't work at all, even for images in my src folder.

Just add your `src` directory to `include` and it'll work again.

By the way, you're repo doesn't work without Firebase or with any Node.js version other than 8 and the error I got was that global CSS isn't supported in files inside node_modules. I didn't see anything related to SVG files. Can you create a repo that actually reproduces the error you're seeing?

**Comment by moaaz-bhnas** [comment]
@lboecker I'm sorry for the late response ..  
First, thank u so much for taking the time to try solve my issue.  

> By the way, you're repo doesn't work without Firebase or with any Node.js version other than 8

My bad. I should've stated that, but this is the first time i write an issue ..  

> the error I got was that global CSS isn't supported in files inside node_modules.

Actually, that was another error that i solved by importing the css in _app.js file, and I dont know a way to reproduce only the image issue without that CSS error.

**Reply by DophinL** [reply]
@moaaz-bhnas hi dude, how you solved this problem? I still meet this in today
